### ISSUE TITLE
No longer works with Stellaris

### ISSUE THREAD

**Issue by deschain212** [ThibautSF/ParadoxosModManagerRework#43]
Stellaris now has a built in "playset" and now when you try applying it with your mod manager it overrides the mod manager config.

**Comment by ThibautSF** [comment]
Can you try the following?:
- Launch the game launcher
- Launch PMM and apply your list
- Launch the game through the launcher (without touching anything else than "play")

**Comment by deschain212** [comment]
Yea I tried that. It just doesnt apply the modlist anymore.

**Comment by deschain212** [comment]
Apparently this was also added for Hearts of Iron IV and it also does not apply to it.

**Comment by fgf1011** [comment]
Hello, I met the same issue. Can you fix that? Thank you.
By the way, Paradox seems use a weird method to build a playlist via a type of file called SQlite. Basically it is a database that builds references list to mods.

**Comment by KevinKevin-2020** [comment]
Same issus here.

The new launcher use SQlite to create mod list and the mod manager is no longer working.

please have a look, and save us from this painful experience of scrolling a hundred mods up and down.

**Comment by fgf1011** [comment]
![6c7077f856aefd6ad681931688bdb3b](https://user-images.githubusercontent.com/46833544/89860049-082e7200-dbd5-11ea-96a6-2100b6391b14.png)
![9507bafcd4c09b0457e0dcaa45fc754](https://user-images.githubusercontent.com/46833544/89860062-14b2ca80-dbd5-11ea-940a-5c9dd450f17c.png)
![8b45fe131ea7bf68a9b799ddf11e549](https://user-images.githubusercontent.com/46833544/89860069-1a101500-dbd5-11ea-8290-f6eff2877d90.png)

I tried to open the SQlite via a sqlitebrowser. However, I never learn CS before so I have no idea.

**Comment by ThibautSF** [comment]
Games affected, all Paradox games using new launcher system.
- Stellaris (tested) 
- Europa Universalis (supposed)
- HoI4 (supposed and reported)
- Imperator Rome (tested)

**Comment by ThibautSF** [comment]
TODO:
- Study file used (looks like to be `launcher-v2.sqlite`) 
- Study database structure
- Find a java library to query sqlite DB
- Find workarounds to apply mods

Workarounds ideas:
- Import lists from playset when PMM start (or as an option like: "Update my lists from Paradox launcher") 
- Apply a list will create a playset (or update the playset with same name)

**Comment by ThibautSF** [comment]
Note:
Launching the game directly by it's executable (might be "stellaris.exe", I'm not sure because I'm not at home and can't check game folder...) should be an option.

- Apply you mod list as usual
- Don't start the game or launcher (through steam, nor PMM, nor GoG...)
- Go in the game's install folder and use the direct executable

**Comment by deschain212** [comment]
I can confirm that the workaround works. Its the way to go for the time being.

**Comment by KarlSchwartz** [comment]
Did exactly what @ThibautSF suggested to fix in HoI 4, with a clean install and few mods, still not working ( sorry for bad english)

**Comment by Crack-Attack** [comment]
Launching the game through the executable did not work for me on Stellaris

**Comment by bcssov** [comment]
@KarlSchwartz @Crack-Attack 
The workaround works it's just that you cannot under any circuimstances open the PDX Launcher as it will instantly overwrite your changes made via PMM. If you do you must reapply the PMM list again.

**Comment by KarlSchwartz** [comment]
@bcssov , I understand what you wrote, but I can assure you, that I was well aware that the launcher would interfere with the list, so I applied the list and launched through the exe, still, no results

**Comment by Crack-Attack** [comment]
@bcssov Yeah nvm the workaround worked for me. I didn't test it properly.

**Comment by mkemper31** [comment]
@ThibautSF Hi, I ran into this same issue and I threw together a solution. I've created a simple JS script that generates a SQL query that can be used to insert into the new sqlite db. At present it only generates the query; I ran it manually through an SQLite DB browser. I will attach it to this message. It is very rough but works - just needs modification to work with your code, make sure it works universally, and make it less hardcoded :) 

Code follows below:

```js
const dlcload = {"enabled_mods":["mod/ugc_2114559224.mod","mod/ugc_2183551683.mod"],"disabled_dlcs":[]};


const generateModDbQuery = (dlcLoadJson) => {
    let i = 0;
    const playsetId = "a3efbe94-2688-421a-8c83-ab3a242ce52e";
    const startValue = parseInt('3333333334', 16);
    let sql = `INSERT INTO playsets_mods VALUES `;
    for (const mod of dlcLoadJson['enabled_mods']) {
        const nextValue = startValue + i;
        if (i) {
            sql += ', ';
        }
        sql += `('${playsetId}', (SELECT id FROM mods WHERE mods.gameRegistryId = '${mod}' LIMIT 1), '${nextValue.toString(16)}', 1)`;
        sql += ';';
        i++;
    }
    return sql;
}

const sql = generateModDbQuery(dlcload);
console.log(sql);
```

**Comment by bcssov** [comment]
@mkemper31
```const playsetId = "a3efbe94-2688-421a-8c83-ab3a242ce52e"```
There's nothing to decode, this is what you call a GUID or UUID.

```const startValue = parseInt('3333333334', 16);```
Well, not exactly a proper hex number you see in my test cases I've seen generated position values "33333333ci" or "333333333m" or "33333333bf" for example. If you decode the number and then sort by it you don't get the proper order which they seem to get. They seem to be sorting these strings alphabetically ;)

I just kinda fetch them from the database sort them the same way they're doing and then update all entries with my own hex numbers. I have no idea what they were thinking to be honest.

**Comment by deschain212** [comment]
Any news on the update?

**Comment by shadoxxhd** [comment]
If the mod manager directly launched Stellaris.exe instead of the paradox launcher, the problem would be mitigated.

Is there a way to make steam launch Stellaris.exe (or PMM) instead of the stupid paradox launcher?

**Comment by bcssov** [comment]
You can create a shortcut on your desktop to the stellaris.exe and launch that, instead of launching via PMM. PMM does not detect the location of your game installs to be able to run the exe directly at the moment I think.
